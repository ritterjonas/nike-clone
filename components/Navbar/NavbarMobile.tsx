import Image from 'next/image';
import {
  Drawer,
  DrawerContainer,
  DrawerItem,
  NavbarComponent,
  NavbarIcon,
  NavbarIcons,
  Overlay,
} from './NavbarMobile.styles';
import { useState } from 'react';

export default function NavbarMobile() {
  const [openDrawer, setOpenDrawer] = useState(false);

  return (
    <>
      <NavbarComponent data-testid='navbar-mobile'>
        <Image
          src='/images/logo_nike.svg'
          alt='Nike Logo'
          width={40}
          height={15}
          data-testid='nike-logo-mobile'
        />

        <NavbarIcons>
          <NavbarIcon href='#'>
            <Image
              src='/images/icon_sacola.svg'
              alt='Bag Icon'
              width={20}
              height={20}
            />
          </NavbarIcon>
          <NavbarIcon href='#' onClick={() => setOpenDrawer(true)}>
            <Image
              src='/images/icon_hamburguer.svg'
              alt='Hamburguer Icon'
              width={24}
              height={24}
            />
          </NavbarIcon>
        </NavbarIcons>
      </NavbarComponent>

      {openDrawer && (
        <DrawerContainer>
          <Drawer data-testid='drawer'>
            <DrawerItem href='#'>Lan√ßamento</DrawerItem>
            <DrawerItem href='#'>Masculino</DrawerItem>
            <DrawerItem href='#'>Feminino</DrawerItem>
            <DrawerItem href='#'>Infantil</DrawerItem>
            <DrawerItem href='#'>Ofertas</DrawerItem>
            <DrawerItem href='#'>SNKRS</DrawerItem>
          </Drawer>
          <Overlay
            onClick={() => setOpenDrawer(false)}
            data-testid='overlay'
          ></Overlay>
        </DrawerContainer>
      )}
    </>
  );
}
