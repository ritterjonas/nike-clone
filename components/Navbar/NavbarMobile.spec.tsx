import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import NavbarMobile from './NavbarMobile';

describe('<NavbarMobile />', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<NavbarMobile />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('opens the drawer when the hamburguer icon is clicked', () => {
    render(<NavbarMobile />);

    const hamburguerIcon = screen.getByAltText('Hamburguer Icon');
    fireEvent.click(hamburguerIcon);

    const drawer = screen.getByTestId('drawer');
    expect(drawer).toBeInTheDocument();
  });

  it('closes the drawer when the overlay is clicked', () => {
    render(<NavbarMobile />);

    const hamburguerIcon = screen.getByAltText('Hamburguer Icon');
    fireEvent.click(hamburguerIcon);

    const overlay = screen.getByTestId('overlay');
    fireEvent.click(overlay);

    const drawer = screen.queryByTestId('drawer');
    expect(drawer).toBeNull();
  });
});
