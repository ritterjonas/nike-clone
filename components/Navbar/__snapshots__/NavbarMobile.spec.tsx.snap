// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarMobile /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-85bd0fd4-0 hdtlb"
    data-testid="navbar-mobile"
  >
    <img
      alt="Nike Logo"
      data-nimg="1"
      data-testid="nike-logo-mobile"
      decoding="async"
      height="15"
      loading="lazy"
      src="/images/logo_nike.svg"
      style="color: transparent;"
      width="40"
    />
    <div
      class="sc-85bd0fd4-1 hszeMi"
    >
      <a
        class="sc-85bd0fd4-2 iCNiDP"
        href="#"
      >
        <img
          alt="Bag Icon"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/images/icon_sacola.svg"
          style="color: transparent;"
          width="20"
        />
      </a>
      <a
        class="sc-85bd0fd4-2 iCNiDP"
        href="#"
      >
        <img
          alt="Hamburguer Icon"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/images/icon_hamburguer.svg"
          style="color: transparent;"
          width="24"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
